pkg/tasks/tasks.go

```diff
@@ -112,24 +113,57 @@ type TaskSpec struct {
 	Task Task
 }
 
-type Task interface {
-	Run(ctx context.Context) error
+type StateErrors = client.StateErrors
```

Comment:
I feel the Task.Run should return a `TaskrunError` which should be 
```go
type TaskResult struct {
  Errors client.StateErrors
  Name string
}
```
instead of it currently returning `client.StateError` which replaced former `Task.Run(ctx) error` 
WDYT?